hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientUtils.java

```diff
@@ -69,9 +71,24 @@ public static BucketLayout resolveLinkBucketLayout(OzoneBucket bucket,
             DETECTED_LOOP_IN_BUCKET_LINKS);
       }
 
-      OzoneBucket sourceBucket =
-          objectStore.getVolume(bucket.getSourceVolume())
-              .getBucket(bucket.getSourceBucket());
+      OzoneBucket sourceBucket;
+      try {
+        sourceBucket =
+            objectStore.getVolume(bucket.getSourceVolume())
+                .getBucket(bucket.getSourceBucket());
+      } catch (OMException ex) {
+        if (ex.getResult().equals(VOLUME_NOT_FOUND)
+            || ex.getResult().equals(BUCKET_NOT_FOUND)) {
+          // for orphan link bucket, return layout as link bucket
+          bucket.setSourcePathExist(false);
+          LOG.error("Source Bucket is not found, its orphan bucket and " +
+              "used link bucket {} layout {}", bucket.getName(),
+              bucket.getBucketLayout());
+          return bucket.getBucketLayout();
```

Comment:
What is the layout of the linked bucket?  Is it the layout of the original source bucket or is it defaulted to some value?

Reply:
Its configured OM default layout "ozone.default.bucket.layout"

Reply:
Thanks.  I ran it the `OzoneClientUtils.resolveLinkBucketLayout` with default, and user specified layouts on bucket create.  No problems - the linked bucket object layout is set to the source bucket type.